### PR TITLE
Glog reinitializes and fails in custom auth plugins

### PR DESCRIPTION
**Pull request by tharindulak**
## Purpose
To avoid glog being initialized again in plugin auth implementations.

## Approach
Separated authn and authz interfaces from authn and authz packages and moved to api package.

## Test environment
go version: go1.12.5 darwin/amd64
OS: Mac OS 10.14.5

### CODE REVIEW

File: auth_server/authn/tokendb.go [hunk 1]
```diff
@@ -22,11 +22,11 @@ import (
 	"fmt"
 	"time"
 
-	"golang.org/x/crypto/bcrypt"
-
+	"github.com/cesanta/docker_auth/auth_server/api"
```
Comment (rojer) [inline auth_server/authn/tokendb.go]: please order imports in 3 groups: system, 3rdparty, then local. this should be at the bottom of the list.

File: auth_server/authn/tokendb_gcs.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ import (
 	"time"
 
 	"cloud.google.com/go/storage"
+	"github.com/cesanta/docker_auth/auth_server/api"
```
Comment (rojer) [inline auth_server/authn/tokendb_gcs.go]: move into its own group at the bottom

File: auth_server/authz/acl.go [hunk 3]
```diff
@@ -10,7 +10,7 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/cesanta/docker_auth/auth_server/authn"
+	"github.com/cesanta/docker_auth/auth_server/api"
```
Comment (rojer) [inline auth_server/authz/acl.go]: same

File: auth_server/authz/plugin_authz.go [hunk 4]
```diff
@@ -20,14 +20,15 @@ import (
 	"fmt"
 	"plugin"
 
+	"github.com/cesanta/docker_auth/auth_server/api"
```
Comment (rojer) [inline auth_server/authz/plugin_authz.go]: same, here and elsewhere

File: auth_server/authn/tokendb.go (hunk 1 above)
Comment (tharindulak) [reply auth_server/authn/tokendb.go]: Fixed

File: auth_server/authn/tokendb_gcs.go (hunk 2 above)
Comment (tharindulak) [reply auth_server/authn/tokendb_gcs.go]: Fixed

File: auth_server/authz/acl.go (hunk 3 above)
Comment (tharindulak) [reply auth_server/authz/acl.go]: Fixed

File: auth_server/authz/plugin_authz.go (hunk 4 above)
Comment (tharindulak) [reply auth_server/authz/plugin_authz.go]: Fixed

**Comment by rojer** [comment]
ok, looks good. can you squash into 1 commit and rebase onto latest master please?

**Comment by tharindulak** [comment]
Squashed into 1 commit

**Pull request merged**